lib/galaxy/webapps/galaxy/api/object_store.py

```diff
@@ -0,0 +1,80 @@
+"""
+API operations on Galaxy's object store.
+"""
+import logging
+from typing import (
+    Any,
+    Dict,
+    List,
+)
+
+from fastapi import (
+    Path,
+    Query,
+)
+
+from galaxy.exceptions import (
+    ObjectNotFound,
+    RequestParameterInvalidException,
+)
+from galaxy.managers.context import ProvidesUserContext
+from galaxy.objectstore import BaseObjectStore
+from . import (
+    depends,
+    DependsOnTrans,
+    Router,
+)
+
+log = logging.getLogger(__name__)
+
+router = Router(tags=["object sstore"])
+
+ConcreteObjectStoreIdPathParam: str = Path(
+    ..., title="Concrete Object Store ID", description="The concrete object store ID."
+)
+
+SelectableQueryParam: bool = Query(
+    default=False, title="Selectable", description="Restrict index query to user selectable object stores."
+)
+
+
+@router.cbv
+class FastAPIObjectStore:
+    object_store: BaseObjectStore = depends(BaseObjectStore)
+
+    @router.get(
+        "/api/object_store",
+        summary="",
+        response_description="",
+    )
+    def index(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        selectable: bool = SelectableQueryParam,
+    ) -> List[Dict[str, Any]]:
+        if not selectable:
+            raise RequestParameterInvalidException(
+                "The object store index query currently needs to be called with selectable=true"
```

Comment:
What's the reason for that ? Seems a little odd. If there's a good reason `SelectableQueryParam` should be marked as mandatory and be the only valid value.

Reply:
This is a narrow view of the object stores and just returns like concrete object stores for selection here and what is needed for selection (pydantic models implemented in the follow up already but namely description, badges, name, etc...). I don't want to think through what a nested store should return - we don't have an application requiring that currently. By having this parameter and making it clear we expect returns to be for selection - we are keeping this open for other applications that might require a more expansive view of the object stores.